C/services/south/south.cpp

```diff
@@ -547,6 +550,22 @@ void SouthService::start(string& coreAddress, unsigned short corePort)
 						}
 						else // V2 poll method
 						{
+							PRINT_FUNC;
+							while(1)
+							{
+								unsigned int numPendingReconfs;
+								{
+									lock_guard<mutex> guard(m_pendingNewConfigMutex);
+									numPendingReconfs = m_pendingNewConfig.size();
+								}
+								if (numPendingReconfs)
+								{
+									Logger::getLogger()->debug("SouthService::start(): %d entries in m_pendingNewConfig, poll thread yielding CPU", numPendingReconfs);
+									std::this_thread::sleep_for(std::chrono::milliseconds(200));
+								}
+								else
+									break;
```

Comment:
Is this the way shutdown gets done in a tidy fashion? If so a comment might be useful.

Reply:
Even now, service shutdown gets done eventually in case of service disable, but may get pended by a few plugin_poll calls. But at foglamp shutdown, it seems services get just 10 seconds for shutdown, else their process is killed by core, which doesn't seem right.